### PR TITLE
Visualize Kafka Connect dead letter topics & Add generic Extractors

### LINKED ISSUE #16: Visualize Kafka Connect dead letter topics
https://docs.confluent.io/platform/current/installation/configuration/connect/sink-connect-configs.html#errors.deadletterqueue.topic.name

### PR DESCRIPTION
**Pull request by disrupted**
Closes #16 #31

### CODE REVIEW

File: backend/streams_explorer/core/services/kafkaconnect.py [hunk 1]
```diff
@@ -29,11 +29,16 @@ def connectors() -> List[KafkaConnector]:
...
+            connector: Optional[
+                KafkaConnector
+            ] = extractor_container.on_connector_config_parsing(info["config"], name)
+            # if no specific connector is returned create generic one
+            if not connector:
+                connector = KafkaConnector(
+                    name=name,
+                    config=info["config"],
+                    type=info["type"],
+                    topics=[],
```
Comment (philipp94831) [inline backend/streams_explorer/core/services/kafkaconnect.py:41]: If it is a sink, we can always read the topics field https://docs.confluent.io/platform/current/installation/configuration/connect/sink-connect-configs.html#sinkconnectorconfigs_topics Same is true for error topics

File: backend/streams_explorer/core/services/kafkaconnect.py [hunk 2]
```diff
@@ -29,11 +29,16 @@ def connectors() -> List[KafkaConnector]:
...
-            topics: List[str] = extractor_container.on_connector_config_parsing(
-                info["config"], name
-            )
-            connector = KafkaConnector(
-                name=name, config=info["config"], type=info["type"], topics=topics
-            )
+            connector: Optional[
+                KafkaConnector
+            ] = extractor_container.on_connector_config_parsing(info["config"], name)
+            # if no specific connector is returned create generic one
```
Comment (philipp94831) [inline backend/streams_explorer/core/services/kafkaconnect.py:35]: What about a test?

**Review by philipp94831 — CHANGES_REQUESTED** [review]

File: backend/streams_explorer/core/services/kafkaconnect.py (hunk 1 above)
Comment (disrupted) [reply backend/streams_explorer/core/services/kafkaconnect.py]: Added new generic extractors for sink & source connectors as fallback. I think this way it is cleaner than putting the sink connector specific parsing in this method.

File: backend/streams_explorer/core/services/kafkaconnect.py (hunk 2 above)
Comment (disrupted) [reply backend/streams_explorer/core/services/kafkaconnect.py]: Yes good point, I am gonna add some tests for the new extractors.

**Review by philipp94831 — APPROVED** [review]

**Pull request merged**